src/store/index.js

```diff
@@ -17,35 +29,49 @@ export default createStore({
     pagination: {
       page: 1,
       size: 15,
-      total: 100,
+      total: 0,
     },
-    formAction: '',
   },
   getters: {
     isLoggedIn: (state) => Boolean(state.token),
     getMessages: (state) => state.messages,
     getPagination: (state) => state.pagination,
-    getFormAction: (state) => state.formAction,
   },
   mutations: {
-    [OPEN_APP_MODAL]: (state, payload) => {
+    [OPEN_APP_MODAL]: (state) => {
       state.isModalActive = true;
-      state.formAction = payload;
     },
     [CLOSE_APP_MODAL]: (state) => {
       state.isModalActive = false;
     },
-    [SET_USER]: (state, { token }) => {
-      state.token = token;
-      localStorage.setItem('token', token);
+    [SET_USER]: (state, payload) => {
+      state.token = payload.token;
+      localStorage.setItem('token', payload.token);
     },
-    [SET_PAGINATION]: (state, { page, total }) => {
-      state.pagination.page = page;
-      state.pagination.total = total;
+    [SET_PAGINATION]: (state, payload) => {
+      state.pagination.page = payload.page;
+      state.pagination.total = payload.total;
     },
     [SET_MESSAGES]: (state, payload) => {
       state.messages = payload;
     },
+    [REMOVE_MESSAGE]: (state, payload) => {
+      state.messages = state.messages.filter((message) => {
+        return message.messageId !== payload;
+      });
+    },
+    [INCREMENT_PAGINATION_TOTAL]: (state) => {
+      state.pagination.total += 1;
+    },
+    [DECREMENT_PAGINATION_TOTAL]: (state) => {
+      state.pagination.total -= 1;
+    },
+    [UPDATE_MESSAGE]: (state, payload) => {
```

Comment:
Use destructuring here
```suggestion
    [UPDATE_MESSAGE]: (state, { id, updatedMessage} ) => {
```